egs/wsj/s5/utils/format_lm_sri.sh

```diff
@@ -104,6 +104,27 @@ awk '{print $1}' $out_dir/words.txt > $tmpdir/voc || exit 1;
 change-lm-vocab -vocab $tmpdir/voc -lm $tmpdir/lm.gz -write-lm $tmpdir/out_lm \
   $srilm_opts || exit 1;
 
+# change-lm-vocab introduces error in out_lm when lm.gz has an order
```

Comment:
Sure, if it's easy enough.

On Fri, Apr 1, 2016 at 8:20 PM, Kirill Katsnelson <notifications@github.com>
wrote:

> In egs/wsj/s5/utils/format_lm_sri.sh
> <https://github.com/kaldi-asr/kaldi/pull/639#discussion_r58283845>:
>
> > @@ -104,6 +104,27 @@ awk '{print $1}' $out_dir/words.txt > $tmpdir/voc || exit 1;
> >  change-lm-vocab -vocab $tmpdir/voc -lm $tmpdir/lm.gz -write-lm $tmpdir/out_lm \
> >    $srilm_opts || exit 1;
> >
> > +# change-lm-vocab introduces error in out_lm when lm.gz has an order
>
> What I can do is ignore highest-order n-grams down until there is an order
> of non-zero cardinality. I think this is what actually happens here. This
> could be useful as it can potentially interfere with the selection of
> packed vs regular history keys. Packed keys support up to 4-grams,
> inclusively. If there is a 0 count 5-gram section, that will pessimize the
> compile.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/639/files/9778cccbca078b8e7e8af1c6ddb7da3b3a1180b8#r58283845>
>